### ISSUE TITLE
find() with paginateOn: 'server' has query ignored if no results returned

### ISSUE THREAD

**Issue by MajesticPotatoe** [marshallswain/feathers-pinia#126]
When using useFind() with { paginateOn: 'server' } and a query is performed via find() (destructured or service().find()) that returns empty results, data doesn't update. When this occurs, the find() query is ignored and will revert to displaying whatever the previous query results (essentially ignoring the query all together). This can be observed via pinia devtools. (queriedAt for the most recent pagination doesn't even update).

This also can be seen indirectly with { paginateOn: 'hybrid' } as well.

**Comment by 1sses** [comment]
Yeah, noticed that too. I think the problem is here, when query is not tracked when `total` is 0
<img width="986" alt="image" src="https://github.com/marshallswain/feathers-pinia/assets/75974774/4f74dd79-a755-450f-811f-095a7c551c33">

**Comment by MajesticPotatoe** [comment]
no worries, pretty much my work around for the time being is to just clear store before query (not ideal but luckily it doesn't effect much for my use case)

**Comment by marshallswain** [comment]
I'm going to update that conditional to see if it fixes our problem.  It's a quick and dirty fix, since I didn't specifically update tests for it.  I've just fixed the conditional in `feathers-pinia@3.0.6`.  Please try it out to see if it fixes the issue you're facing.  Thanks again for reporting!

**Comment by marshallswain** [comment]
I'm reverting the change.  Integrating into a published app fails.  I'll work on this another time.

**Comment by laurensiusadi** [comment]
@marshallswain I experienced this bug too, and I think I have the fix

https://github.com/marshallswain/feathers-pinia/blob/main/src/hooks/3-sync-store.ts#L31

The `syncStore` hook skips `updatePaginationForQuery()` when the find response has `total === 0` because `0` is falsy

```javascript
      // Update pagination based on the qid
      if (method === 'find' && context.result.total) {
        const { qid = 'default', query, preserveSsr = false } = context.params
        store.updatePaginationForQuery({ qid, response: context.result, query, preserveSsr })
      }
```
When a service returns empty results (e.g., a new table with no rows), the pagination entry is never created in the Pinia store. This means:

1. **Server**: Fetches data, gets `{ data: [], total: 0 }`, but no pagination entry is stored. The `ssr` flag is never set.
2. **Client**: Pinia hydrates the store from SSR payload, but there's no pagination data to hydrate. `useFind` sees no SSR data, falls through to `makeRequest()`, which sets `isPending = true`.
3. **Mismatch**: Server rendered with `isPending = false` (data loaded, just empty), but client hydrates with `isPending = true` (re-fetching). Any UI bound to `isPending` (loading spinners, skeleton states, CSS classes) will differ.

### Fix

```diff
- if (method === 'find' && context.result.total) {
+ if (method === 'find' && context.result.total != null) {
```

Use `!= null` (loose equality) to allow `total: 0` while still guarding against `undefined` (non-paginated responses).
